nilearn/image/resampling.py

```diff
@@ -340,6 +339,8 @@ def resample_img(img, target_affine=None, target_shape=None,
     This function handles gracefully NaNs and infinite values in the input
     data, however they make the execution of the function much slower.
     """
+    from .image import new_img_like
```

Comment:
> So we stay in _utils with new_img_like, 

We can try doing the internal imports from image.image to see if it
solves the problem (add a comment saying that this is to avoid circular
imports).